daos/story-dao.ts

```diff
@@ -79,16 +84,18 @@ export default class StoryDao implements StoryDaoI {
    * database
    */
   findStories = async (): Promise<Story[]> => {
-    return StoryModel.find().populate("postedBy");
+    return StoryModel.find().populate("viewers").populate("postedBy").exec();
 }
 
   /**
    * Retrieves all stories by visibility, as stories can be visible to public, friends or
    * close friends groups
-   * @param visibility Visibility of story
+   * @param uid
    */
-  findStoriesByVisibility = async (visibility: string): Promise<Story[]> => {
-    return StoryModel.find({visibility: visibility}).populate("postedBy");
+  findStoriesVisibleToUser = async (uid: string): Promise<Story[]> => {
+    const stories = await this.findStories();
+    const user = await StoryDao.userDao.findUserById(uid);
+    return stories.filter((story) => (story.viewers.length < 1 ||
```

Comment:
Since `viewers` is not a `required` parameter in mongoose schema, and we don't have a default value of `[]` empty array for it, we might also need to check if the `viewers` field is present in `Story`. Maybe we can have the filter like `stories.filter((story) => (!story.viewers || story.viewers.length < 1 || story.viewers.find(viewer => String(viewer._id)===uid)))`

Reply:
Is it ok if instead of checks I add a default empty list in mongoose schema?

Reply:
Yes that works!